aixtend/processes/data_onboarding/process_audio_files.py

```diff
@@ -0,0 +1,195 @@
+__author__ = "thiagocastroferreira"
+
+import logging
+import os
+import pandas as pd
+import shutil
+import tarfile
+
+from aixtend.enums.file_type import FileType
+from aixtend.enums.storage_type import StorageType
+from aixtend.modules.file import File
+from aixtend.modules.metadata import MetaData
+from aixtend.utils.file_utils import upload_data
+from pathlib import Path
+from tqdm import tqdm
+from typing import List, Tuple
+
+
+def compress_folder(folder_path: str):
+    with tarfile.open(folder_path + ".tgz", "w:gz") as tar:
+        for name in os.listdir(folder_path):
+            tar.add(os.path.join(folder_path, name))
+    return folder_path + ".tgz"
+
+
+def run(metadata: MetaData, paths: List, folder: Path, batch_size: int = 100) -> Tuple[List[File], int, int, int]:
+    """Process a list of local audio files, compress and upload them to pre-signed URLs in S3
+
+    Args:
+        metadata (MetaData): meta data of the asset
+        paths (List): list of paths to local files
+        folder (Path): local folder to save compressed files before upload them to s3.
```

Comment:
What is batch size used for? How is the data stored in batches?
A supporting document or comment can help implementors understand the code here.